src/Core/WinSWCore/ServiceDescriptor.cs

```diff
@@ -85,6 +85,43 @@ public ServiceDescriptor()
             Environment.SetEnvironmentVariable(WinSWSystem.ENVVAR_NAME_SERVICE_ID, Id);
         }
 
+        //This constructor is called when config file specified in command line
+        public ServiceDescriptor(string filepath)
+        {
+            BaseName = "";
+            BasePath = "";
+
+            if (filepath.EndsWith(".xml"))
+                filepath = filepath.Substring(0, filepath.Length - 4);

```

Comment:
Why is the extension stripped?

Reply:
Someone may specify file path with extension and someone may not. To avoid that conflict. Also as in XML schema validation XSD file is also have same name as XML file. So have to make sure load the XML file.

Reply:
Are you expecting the user to provide the XSD file?

Reply:
No. what I did is find the xsd file with the same name of xml file

Reply:
Firstly, why do you pass to trimmed file path to `File.Exists`?

Reply:
Ooh I missed that. That is why File.Exist() method return false. If user specify the complete path with extension then this will be okay. I was thinking user will specify only the file name without extension. Thank you for mention this.